File: leetcode/solutions/1009.ComplementOfBase10Integer/bitwiseComplement.ts
export default function bitwiseComplement(N: number): number {
  if (N === 0) {
    return 1;
  }

  let base2 = "";

  for (; N > 0; N = Math.floor(N / 2)) {
    base2 = (N % 2 ^ 1).toString() + base2;
  }

  return parseInt(base2, 2);
}

File: leetcode/solutions/1009.ComplementOfBase10Integer/bitwiseComplement.go
package solution

import "math"

func bitwiseComplement(N int) int {
	if N == 0 {
		return 1
	}

	digits := math.Floor(math.Log2(float64(N))) + 1
	return int((math.Pow(2, digits) - 1)) ^ N
}
